docs/resources/asserts_custom_model_rules.md

```diff
@@ -0,0 +1,108 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "grafana_asserts_custom_model_rules Resource - terraform-provider-grafana"
+subcategory: "Asserts"
+description: |-
+  Manages Asserts Custom Model Rules through the Grafana API.
+---
+
+# grafana_asserts_custom_model_rules (Resource)
+
+Manages Asserts Custom Model Rules through the Grafana API.
+
+## Example Usage
+
+```terraform
+resource "grafana_asserts_custom_model_rules" "test" {
+  name = "test-anything"
+  rules {
+    entity {
+      type = "Service"
+      name = "workload | service | job"
+      scope = {
+        namespace = "namespace"
+        env       = "asserts_env"
+        site      = "asserts_site"
+      }
+      lookup = {
+        workload  = "workload | deployment | statefulset | daemonset | replicaset"
+        service   = "service"
+        job       = "job"
+        proxy_job = "job"
+      }
+      defined_by {
+        query    = "up{job!=''}"
+        disabled = false
+        label_values = {
+          service = "service"
+          job     = "job"
+        }
+        literals = {
+          _source = "up_query"
+        }
+      }
+      defined_by {
+        query    = "up{job='disabled'}"
+        disabled = true
+      }
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The name of the custom model rules.
+- `rules` (Block List, Min: 1, Max: 1) The rules configuration for the custom model rules. (see [below for nested schema](#nestedblock--rules))
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+
+<a id="nestedblock--rules"></a>
+### Nested Schema for `rules`
+
+Required:
+
+- `entity` (Block List, Min: 1) List of entities to define in the custom model rules. (see [below for nested schema](#nestedblock--rules--entity))
+
+<a id="nestedblock--rules--entity"></a>
+### Nested Schema for `rules.entity`
+
+Required:
+
+- `defined_by` (Block List, Min: 1) List of queries that define this entity. (see [below for nested schema](#nestedblock--rules--entity--defined_by))
+- `name` (String) The name of the entity.
+- `type` (String) The type of the entity (e.g., Service, Pod, Namespace).
+
+Optional:
+
+- `disabled` (Boolean) Whether this entity is disabled.
+- `enriched_by` (List of String) List of enrichment sources for the entity.
```

Comment:
`enriched_by` has the same schema as `defined_by`.
It's a [`PropertyRule`](https://github.com/grafana/asserts-adi/blob/master/model/rules/src/main/java/ai/asserts/model/rules/PropertyRule.java)